### PR TITLE
fix(chartgen): prefer registry-aware image paths

### PR DESCRIPTION
**Pull request by omercnet**
## Summary
- Prefer image value paths that declare `registry`/`defaultRegistry` when duplicate parent/subchart paths match the same repository.
- Add a regression case for Airflow's parent PostgreSQL image values versus bundled Bitnami PostgreSQL subchart values.

## Validation
- `go test ./internal/chartgen ./internal/discovery`
- `go test ./...`
- `PATH=\"$(go env GOPATH)/bin:$PATH\" make lint`

## Context
After [#411](https://github.com/verity-org/verity/pull/411), main Chart Generation still produced an Airflow wrapper with `repository: ghcr.io/verity-org/bitnamilegacy/postgresql` and an empty registry because chart-gen selected the parent chart path before the registry-aware subchart path. Main chart-integration verified `mimir-distributed` successfully, but Airflow remained red until this path selection is corrected.

---
## Summary by cubic
Prefer registry-aware image value paths and ensure each mapping uses a unique image path when duplicates exist. Fixes Airflow selecting the parent PostgreSQL path with an empty registry by choosing the subchart path that declares `image.registry`/`global.imageRegistry`.

- **Bug Fixes**
  - Select the best repo/tag pair per mapping and consume duplicates: prefer paths with `image.registry` or `global.imageRegistry`, then deeper, then longer.
  - Prevent empty or duplicated registries (e.g., avoid ghcr.io/ghcr.io).
  - Add regression tests for Airflow and duplicate repo pairs.

<sup>Written for commit 6039f079bb8142197bed17d3e02f4fa58d09e7d3. Summary will update on new commits. <a href="https://cubic.dev/pr/verity-org/verity/pull/413?utm_source=github">Review in cubic</a></sup>

### CODE REVIEW

File: internal/chartgen/values.go [hunk 1]
```diff
@@ -162,31 +162,50 @@ func resolveValuePathPairs(pairs []repoTagPair, globalRegistryPaths []string, ma
...
+			best.HasRegistry,
+			best.HasDefaultRegistry,
+		))
+		matched[i] = true
 	}
 
 	return appendGlobalRegistryNeutralisations(result, globalRegistryPaths)
 }
 
+func bestRepoTagPair(pairs []repoTagPair, originalRepo string) *repoTagPair {
```
[the remark this answers is not in the document]
Comment (omercnet) [reply internal/chartgen/values.go]: Fixed in the latest push: `bestRepoTagPair` now tracks consumed pair indexes, so duplicate repository mappings cannot collapse onto the same values path. Added a regression test covering two mappings for the same repo.

File: internal/chartgen/values.go [hunk 2]
```diff
@@ -162,31 +162,50 @@ func resolveValuePathPairs(pairs []repoTagPair, globalRegistryPaths []string, ma
...
 		}
+		result = append(result, buildValueOverride(
+			best.Path,
+			m.PatchedRepo,
+			m.PatchedTag,
+			best.HasRegistry,
+			best.HasDefaultRegistry,
+		))
+		matched[i] = true
 	}
```
[the remark this answers is not in the document]
Comment (omercnet) [reply internal/chartgen/values.go:183]: Fixed in the latest push by tracking selected pair indexes and excluding them from later mapping selection; each values path can now be consumed at most once.

File: internal/chartgen/values.go [hunk 3]
```diff
@@ -162,31 +162,50 @@ func resolveValuePathPairs(pairs []repoTagPair, globalRegistryPaths []string, ma
...
+	return best
+}
+
+func preferRepoTagPair(candidate, current *repoTagPair) bool {
+	candidateHasRegistry := candidate.HasRegistry || candidate.HasDefaultRegistry
+	currentHasRegistry := current.HasRegistry || current.HasDefaultRegistry
+	if candidateHasRegistry != currentHasRegistry {
+		return candidateHasRegistry
+	}
+	return len(candidate.Path) > len(current.Path)
```
[the remark this answers is not in the document]
Comment (omercnet) [reply internal/chartgen/values.go:215]: Fixed in the latest push: the specificity tie-breaker now compares dotted path depth first, with length only as a final deterministic fallback.
